### PR TITLE
V2.9.0 compass

### PR DESCRIPTION
**Pull request by brentpicasso**
Re-opening PR, rebased against v2.9.0

### CODE REVIEW

File: include/logger/loggerConfig.h [hunk 1]
```diff
@@ -212,19 +212,16 @@ typedef struct _ImuConfig {
...
-#define DEFAULT_ACCEL_X_CONFIG    {"AccelX",   "G", -3, 3, SAMPLE_25Hz, 2, 0}
-#define DEFAULT_ACCEL_Y_CONFIG    {"AccelY",   "G", -3, 3, SAMPLE_25Hz, 2, 0}
-#define DEFAULT_ACCEL_Z_CONFIG    {"AccelZ",   "G", -3, 3, SAMPLE_25Hz, 2, 0}
-#define DEFAULT_GYRO_YAW_CONFIG   {"Yaw",      "Deg/Sec", -300, 300, SAMPLE_25Hz, 1, 0}
-#define DEFAULT_GYRO_PITCH_CONFIG {"Pitch",    "Deg/Sec", -300, 300, SAMPLE_25Hz, 1, 0}
-#define DEFAULT_GYRO_ROLL_CONFIG  {"Roll",     "Deg/Sec", -300, 300, SAMPLE_25Hz, 1, 0}
 
 #define DEFAULT_IMU_CHANNEL_CONFIG  {"", "G", -3, 3, SAMPLE_25Hz, 2, 0}
 #define DEFAULT_GYRO_CHANNEL_CONFIG {"", "Deg/Sec", -300, 300, SAMPLE_25Hz, 1, 0}
+#define DEFAULT_COMPASS_CHANNEL_CONFIG {"", "Deg", 0, 360, SAMPLE_DISABLED, 0, 0}
```
Comment (stieg) [inline include/logger/loggerConfig.h]: So not named "Compass" yet?

File: src/logger/loggerConfig.c [hunk 2]
```diff
@@ -92,14 +92,16 @@ static void resetTimerConfig(TimerConfig cfg[])
 
 static void resetImuConfig(ImuConfig cfg[])
 {
-    const char *imu_names[] = {"AccelX", "AccelY", "AccelZ", "Yaw", "Pitch", "Roll"};
+    const char *imu_names[] = {"AccelX", "AccelY", "AccelZ", "Yaw", "Pitch", "Roll", "Compass"};
```
Comment (stieg) [inline src/logger/loggerConfig.c]: Huh... this is... confusing.

File: stm32_base/hal/imu_stm32/invensense_9150.c [hunk 3]
```diff
@@ -2,46 +2,214 @@
...
-    int res = 0;
+    int res;
     res = i2c_read_reg8(is9150_dev.i2c, is9150_dev.addr,
                         reg_addr, reg_val);
     return res;
 }
 
+static int is9150_write_reg(uint8_t reg_addr, uint8_t reg_val)
+{
+    int res;
```
Comment (stieg) [inline stm32_base/hal/imu_stm32/invensense_9150.c]: Nit feedback.  Do
```C
const int res = i2c_write_reg8(is9150_dev.i2c, is9150_dev.addr, reg_addr, reg_val);
```
instead as the compiler can optimize that better and you can still use it in debug statements as needed.

Comment (stieg) [inline stm32_base/hal/imu_stm32/invensense_9150.c]: Applies throughout.

**Comment by stieg** [comment]
Only minor stuff.  Fix if you deem worthy and :shipit:

File: stm32_base/hal/imu_stm32/invensense_9150.c (hunk 3 above)
Comment (brentpicasso) [inline stm32_base/hal/imu_stm32/invensense_9150.c]: thanks. applied feedback.

**Pull request merged**